ClinGen/clincoded
# Connects to #671. UI framework for VCI hub.

This PR is for the preliminary UI framework used by the VCI hub (or variant-central). Test data has been used during the development.

**Technical notes:**
1. I added 2 npm packages in this PR -- `react-tabs` and `react-localStorage`. The use of `react-tabs` allows me to quickly configure the VCI tabs per the mockups without spending time to write our own version of tabs. And the use of `react-localStorage` is meant to store fetched data in the browser's local storage so that persistent data can be used when the user cycle through the tabs.
2. Furthermore, I adopted a file structure in which some JS files are grouped in a subdirectory. Doing so helps me to organize the modules by their context and types. It also gives me the flexibility to name files that might have conflicted with other existing files outside of the subdirectory.

**Steps to test this PR:**
1. Login and go to http://localhost:6543/variant-central/?variant=35eedfdc-2685-11e5-b6a0-60f81dc5b05a
2. Click on various "link-out" links in the header and expect a new window being opened for the external sites (the URLs of certain external sites have not been finalized).
3. Click on the "View" links in the header and expect JSON representation of data being returned.
4. The "Edit" link in the header is not yet functioning as expected.
5. Click on various tabs and expect different content being presented from one tab to another. And also expect to see NO form elements being displayed in the "Population" tab.
6. Click on the "Start new interpretation" button above the tabs and expect to see the "Variant Interpretation Record" title string being displayed, as well as the "Associate with Disease" button on the right and the "Return to Evidence Only" button on the left. Also expect to see form elements being displayed in the "Population" tab.
